src/storage.ts

```diff
@@ -413,10 +414,42 @@ export class Storage extends Service {
     // Note: EMULATOR_HOST is an experimental configuration variable. Use apiEndpoint instead.
     const baseUrl = EMULATOR_HOST || `${options.apiEndpoint}/storage/v1`;
 
+    /**
+     * Returns true if the API request should be retried, given the error that was
+     * given the first time the request was attempted.
+     *
+     * @param {error} err - The API error to check if it is appropriate to retry.
+     * @return {boolean} True if the API request should be retried, false otherwise.
+     */
+    const retryFunction = function (err?: ApiError) {
+      if (err) {
+        if ([408, 429, 500, 502, 503, 504].indexOf(err.code!) !== -1) {
+          return true;
+        }
+
+        if (err.errors) {
+          for (const e of err.errors) {
+            const reason = e.reason;
+            if (
+              reason === 'rateLimitExceeded' ||
+              reason === 'userRateLimitExceeded' ||
+              (reason && reason.includes('EAI_AGAIN')) ||
+              reason === 'Connection Reset By Peer' ||
+              reason === 'Unexpected Connection Closure'
```

Comment:
@frankyn What would be a good way to force these failures to see if this is the correct string to be looking for?

Reply:
Emulator has Reset connection and Unexpected connection closure support. We can talk through this in our meeting later today.